atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyUnicastService.java

```diff
@@ -113,15 +113,18 @@ public void unicast(final Address address, final String subject, final byte[] pa
         .resolve(address.socketAddress())
         .addListener(
             resolvedAddress -> {
+              var cleanUpFuture = group.next().newSucceededFuture((Void) null);
```

Comment:
💭  Given https://github.com/camunda/zeebe/pull/15300#pullrequestreview-1740309725 &  https://github.com/camunda/zeebe/pull/15300#issuecomment-1819496639, we could avoid using the future and "just" do
```
              if (resolvedAddress.isSuccess()) {
                channel.writeAndFlush(
                    new DatagramPacket(buf, (InetSocketAddress) resolvedAddress.get()));
              } else {
                // free allocated buffer if we can't send the message
                buf.release();
                log.warn(
                    "Failed sending unicast message (destination address {} cannot be resolved)",
                    address,
                    resolvedAddress.exceptionNow());
              }
```

Reply:
Which would mean we would rely on netty cleaning up. Idk whether we want this or not, feels somehow not correct. But yeah idk. @npepinpe any opinion?

Reply:
It's part of the Netty contract to clean it up. I mean, we don't do it on the TCP side either :shrug: I think it's safe to rely on them - you wouldn't want to clean it up too early or something without knowing internals.

imo this is a problem only because we use this channel directly instead of using a pipeline :shrug: Normally you would write your message directly (not encoded), and have a outgoing pipeline step which does the encoding at the end, thus ensuring the buffer wouldn't get allocated if any of the previous steps failed. But that's more refactoring, and not sure if we get so much value out of it.

Reply:
@npepinpe sorry for the follow up question, but is this also guaranteed when the writeAndFlush fails? This was actually the reason why I added the future, to handle this as well.